src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -519,30 +519,32 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
             [NotNull] PositionTagged<string> suffix,
             params AttributeValue[] values)
         {
+            bool useAttributeNameAsValue;
+
             if (values.Length == 0)
             {
                 // Explicitly empty attribute, so write the prefix
                 WritePositionTaggedLiteral(writer, prefix);
             }
-            else if (values.Length == 1 && values[0].Prefix == "" &&
-                (values[0].Value.Value is bool || values[0].Value.Value == null))
+            else if (!ShouldRenderAttribute(values, out useAttributeNameAsValue))
+            {
+                return;
+            }
+            else if (useAttributeNameAsValue)
             {
-                // Value is either null or a bool with no prefix.
                 var attributeValue = values[0];
                 var positionTaggedAttributeValue = attributeValue.Value;
 
-                if (positionTaggedAttributeValue.Value == null || !(bool)positionTaggedAttributeValue.Value)
-                {
-                    // The value is null or just the bool 'false', don't write anything.
-                    return;
-                }
-
                 WritePositionTaggedLiteral(writer, prefix);
 
                 var sourceLength = suffix.Position - positionTaggedAttributeValue.Position;
+                var nameAttributeValue = new AttributeValue(
+                    attributeValue.Prefix,
+                    new PositionTagged<object>(name, attributeValue.Value.Position),
+                    literal: attributeValue.Literal);
```

Comment:
So the method signature can be used for both `TagHelper`s and non-`TagHelper`s. Otherwise you'd be passing in bs data from the `TagHelper` side; can be misleading.